protocols/rendezvous/examples/discover.rs

```diff
@@ -53,62 +57,77 @@ async fn main() {
 
     let _ = swarm.dial_addr(rendezvous_point_address.clone());
 
-    while let Some(event) = swarm.next().await {
-        match event {
-            SwarmEvent::ConnectionEstablished { peer_id, .. } if peer_id == rendezvous_point => {
-                log::info!(
-                    "Connected to rendezvous point, discovering nodes in '{}' namespace ...",
-                    NAMESPACE
-                );
+    let mut discover_tick = tokio::time::interval(Duration::from_secs(30));
+    let mut cookie = None;
 
+    loop {
+        tokio::select! {
```

Comment:
`futures::select` would be more generic, presumably this is just a reexport

Reply:
It is not, the implementation is in fact very different: https://github.com/tokio-rs/tokio/commit/8cf98d694663e8397bfc337e7a4676567287bfae

Reply:
It's in my toolbox right now. Didn't bother trying to be runtime agnostic, as the example already used stuff from tokio.